### ISSUE TITLE
Ubuntu crash if closing game window with opponent's hand revealed.

### ISSUE THREAD

**Issue by mcallahan** [Cockatrice/Cockatrice#31]
I crash if I close a game window with the opponent's hand revealed.

Here's what I get with a debug build:

```
GameScene::removePlayer
CardZone destructor:  "deck" 
CardZone destructor:  "grave" 
CardZone destructor:  "hand" 
CardZone destructor:  "rfg" 
CardZone destructor:  "sb" 
CardZone destructor:  "stack" 
CardZone destructor:  "table" 
Player destructor: "muttuser" 
GameScene::removePlayer
CardZone destructor:  "deck" 
CardZone destructor:  "grave" 
CardZone destructor:  "hand" 
CardZone destructor:  "rfg" 
CardZone destructor:  "sb" 
CardZone destructor:  "stack" 
CardZone destructor:  "table" 
ZoneViewZone destructor
CardZone destructor:  "hand" 
*** Error in `/home/callahan/cockatrice-new/Cockatrice/build/cockatrice/./cockatrice': corrupted double-linked list: 0x00000000036cc350 ***
#0  0x00007ffff5347c2b in malloc_consolidate (av=av@entry=0x7ffff5689740 <main_arena>) at malloc.c:4094
#1  0x00007ffff53487e8 in _int_free (av=0x7ffff5689740 <main_arena>, p=0x3c3ed80, have_lock=0) at malloc.c:3994
#2  0x00007ffff769957c in QGraphicsViewPrivate::~QGraphicsViewPrivate() () from /usr/lib/x86_64-linux-gnu/libQtGui.so.4
#3  0x00007ffff683b8e5 in QObject::~QObject() () from /usr/lib/x86_64-linux-gnu/libQtCore.so.4
#4  0x00007ffff70e6bcc in QWidget::~QWidget() () from /usr/lib/x86_64-linux-gnu/libQtGui.so.4
#5  0x00007ffff7690f65 in QGraphicsView::~QGraphicsView() () from /usr/lib/x86_64-linux-gnu/libQtGui.so.4
#6  0x000000000067d699 in GameView::~GameView (this=0x3b74950, __in_chrg=<optimized out>)
    at /home/linuxguy/cockatrice-new/Cockatrice/build/cockatrice/src/../../../cockatrice/src/gameview.h:8
#7  0x000000000067d6d2 in GameView::~GameView (this=0x3b74950, __in_chrg=<optimized out>)
    at /home/linuxguy/cockatrice-new/Cockatrice/build/cockatrice/src/../../../cockatrice/src/gameview.h:8
#8  0x00007ffff6839308 in QObjectPrivate::deleteChildren() () from /usr/lib/x86_64-linux-gnu/libQtCore.so.4
#9  0x00007ffff70e6b37 in QWidget::~QWidget() () from /usr/lib/x86_64-linux-gnu/libQtGui.so.4
#10 0x000000000062ab90 in Tab::~Tab (this=0x3b73a20, __in_chrg=<optimized out>)
    at /home/linuxguy/cockatrice-new/Cockatrice/cockatrice/src/tab.h:10
#11 0x0000000000632d52 in TabGame::~TabGame (this=0x3b73a20, __in_chrg=<optimized out>)
    at /home/linuxguy/cockatrice-new/Cockatrice/cockatrice/src/tab_game.cpp:509
#12 0x0000000000632eaa in TabGame::~TabGame (this=0x3b73a20, __in_chrg=<optimized out>)
    at /home/linuxguy/cockatrice-new/Cockatrice/cockatrice/src/tab_game.cpp:522
#13 0x00007ffff683ae08 in QObject::event(QEvent*) () from /usr/lib/x86_64-linux-gnu/libQtCore.so.4
#14 0x00007ffff70ea36b in QWidget::event(QEvent*) () from /usr/lib/x86_64-linux-gnu/libQtGui.so.4
#15 0x00007ffff709adfc in QApplicationPrivate::notify_helper(QObject*, QEvent*) ()
   from /usr/lib/x86_64-linux-gnu/libQtGui.so.4
#16 0x00007ffff70a1470 in QApplication::notify(QObject*, QEvent*) () from /usr/lib/x86_64-linux-gnu/libQtGui.so.4
#17 0x00007ffff68228bd in QCoreApplication::notifyInternal(QObject*, QEvent*) ()
   from /usr/lib/x86_64-linux-gnu/libQtCore.so.4
#18 0x00007ffff6825e1f in QCoreApplicationPrivate::sendPostedEvents(QObject*, int, QThreadData*) ()
   from /usr/lib/x86_64-linux-gnu/libQtCore.so.4
#19 0x00007ffff6850073 in ?? () from /usr/lib/x86_64-linux-gnu/libQtCore.so.4
#20 0x00007ffff49a23b6 in g_main_context_dispatch () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
#21 0x00007ffff49a2708 in ?? () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
#22 0x00007ffff49a27ac in g_main_context_iteration () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
#23 0x00007ffff684fa55 in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) ()
   from /usr/lib/x86_64-linux-gnu/libQtCore.so.4
#24 0x00007ffff713c9d6 in ?? () from /usr/lib/x86_64-linux-gnu/libQtGui.so.4
#25 0x00007ffff68215ef in QEventLoop::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) ()
   from /usr/lib/x86_64-linux-gnu/libQtCore.so.4
#26 0x00007ffff68218e5 in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) ()
   from /usr/lib/x86_64-linux-gnu/libQtCore.so.4
#27 0x00007ffff6826e5b in QCoreApplication::exec() () from /usr/lib/x86_64-linux-gnu/libQtCore.so.4
#28 0x00000000005acac8 in main (argc=1, argv=0x7fffffffdee8)
    at /home/linuxguy/cockatrice-new/Cockatrice/cockatrice/src/main.cpp:161
```

Interestingly the following patch fixes me.  However I am not sure it is sufficient or why it is necessary:

```
diff --git a/cockatrice/src/gamescene.cpp b/cockatrice/src/gamescene.cpp
index ce083c7..da665a5 100644
--- a/cockatrice/src/gamescene.cpp
+++ b/cockatrice/src/gamescene.cpp
@@ -24,6 +24,7 @@ GameScene::GameScene(PhasesToolbar *_phasesToolbar, QObject *parent)
 
 GameScene::~GameScene()
 {
+       clearViews();
        delete animationTimer;
 }
 
```

**Comment by ZeldaZach** [comment]
Does this issue still occur?

**Comment by mcallahan** [comment]
I have not tested it in a while.  Dunno.

**Comment by ZeldaZach** [comment]
Can you try it out when you have time? :)

**Comment by mcallahan** [comment]
I will be out of town for three weeks.  I will take a look when I get back.

**Comment by ZeldaZach** [comment]
No problem, thanks for your continued support!

**Comment by ctrlaltca** [comment]
Dup #808, fixed in #951
